client/my-sites/stats/stats-purchase/stats-purchase-personal.tsx

```diff
@@ -227,27 +239,59 @@ const PersonalPurchase = ( {
 					{ translate( 'Continue with Jetpack Stats for free' ) }
 				</ButtonComponent>
 			) : (
-				<ButtonComponent
-					variant="primary"
-					primary={ isWPCOMSite ? true : undefined }
-					onClick={ () =>
-						gotoCheckoutPage( {
-							from,
-							type: 'pwyw',
-							siteSlug,
-							adminUrl,
-							redirectUri,
-							price: subscriptionValue / MIN_STEP_SPLITS,
-						} )
-					}
-				>
-					{ continueButtonText }
-				</ButtonComponent>
+				<StatsPurchaseActions
+					isWPCOMSite={ isWPCOMSite }
+					buttonText={ continueButtonText }
+					onCheckoutClick={ handleCheckoutRedirect }
+					onPostponeClick={ handleCheckoutPostponed }
+				/>
 			) }
 		</div>
 	);
 };
 
+function StatsPurchaseActions( props: any ) {
```

Comment:
Please avoid adding `any`. It should be used as a last resort because it beats the purpose of having TS.

Reply:
ha I thought `any` would break type checks but it didn't

Reply:
I actually meant to address this before requesting review. Thanks for the nudge/reminder though. I'll get it fixed up!